pkg/datasource.go

```diff
@@ -721,161 +756,188 @@ func (o *OCIDatasource) processLogMetrics(ctx context.Context, searchLogsReq Gra
 		o.loggingSearchClient.SetRegion(string(reg))
 
 		// Perform the logs search operation
-		res, err := o.loggingSearchClient.SearchLogs(ctx, request)
-
-		if err != nil {
-			o.logger.Debug(fmt.Sprintf("Log search operation FAILED, panelId = %s, refId = %s, err = %s",
-				queryPanelId, queryRefId, err))
-			return errors.Wrap(err, "error fetching logs")
-		}
-		o.logger.Debug("Log search operation SUCCEEDED", "panelId", queryPanelId, "refId", queryRefId,
-			"interval", intervalCnt)
+		// res, err := o.loggingSearchClient.SearchLogs(ctx, request)
```

Comment:
As mentioned on Slack, the visualization of log-based metrics is not working with these changes. And more importantly I don't feel like these changes are really needed as even 500 results of metric data on a single data panel is going to be completely unusable, let alone 10,000.

I would suggest that we not put the pagination logic in the processLogMetrics() method for now. Down the road if there is a clear use case or request that comes from a user to have more than 500 metrics shown on a single data panel then we can look at adding the pagination logic

Reply:
I removed the pagination from this specific function